datadog-setup.php

```diff
@@ -400,6 +701,118 @@ function ($el) {
     }
 }
 
+/**
+ * Returns a list of all INI files found for the `$phpProperties` given.
+ *
+ * Does so by scanning the INI scan directory for `.ini` files. Additionally we
+ * check if we are running on a Debian based distribution so we assume the INI
+ * files are split by SAPI, so we try and add the Apache SAPI INI files as
+ * well. In case it exists, we also add the default `php.ini` file to the list.
+ *
+ * The returned array is somewhat sorted to have the "default" INI file for
+ * Datadog (`98-ddtrace.ini`) in the beginning of the array.
+ *
+ * @see ini_values
+ * @return string[]
+ */
+function find_all_ini_files(array $phpProperties): array
+{
+    $iniFilePaths = [];
+
+    $addIniFiles = function (string $path) use (&$iniFilePaths) {
+        if (!is_dir($path)) {
+            return;
+        }
+        foreach (scandir($path) as $ini) {
+            if (!is_file($path . '/' . $ini) || substr($ini, -4) !== '.ini') {
+                continue;
+            }
+            $iniFile = $path . '/' . $ini;
+            if (strpos($ini, '98-ddtrace.ini') !== false) {
+                array_unshift($iniFilePaths, $iniFile);
+            } else {
+                $iniFilePaths[] = $iniFile;
+            }
+        }
+    };
+
+    if ($phpProperties[INI_SCANDIR]) {
+        $addIniFiles($phpProperties[INI_SCANDIR]);
+
+        if (strpos($phpProperties[INI_SCANDIR], '/cli/conf.d') !== false) {
+            /* debian based distros have INI folders split by SAPI, in a predefined way:
+             *   - <...>/cli/conf.d       <-- we know this from php -i
+             *   - <...>/apache2/conf.d   <-- we derive this from relative path
+             *   - <...>/fpm/conf.d       <-- we derive this from relative path
+             */
+            $apacheConfd = str_replace('/cli/conf.d', '/apache2/conf.d', $phpProperties[INI_SCANDIR]);
+            $addIniFiles($apacheConfd);
+        }
+    }
+
+    if ($phpProperties[INI_MAIN] && is_file($phpProperties[INI_MAIN])) {
```

Comment:
I got this error on one of my configurations:
> Warning: Undefined array key "Loaded Configuration File" in %s/datadog-setup.php on line 753

I assume this was supposed to have `isset()` around it or something?